ballerina/types.bal

```diff
@@ -25,7 +25,8 @@ public type ConnectionConfig record {|
     # The Salesforce endpoint URL
     string baseUrl;
     # Configurations related to client authentication
-    http:BearerTokenConfig|config:OAuth2RefreshTokenGrantConfig auth;
+    http:BearerTokenConfig|config:OAuth2RefreshTokenGrantConfig|config:OAuth2PasswordGrantConfig|
```

Comment:
Is there any particular reason to use the direct HTTP config type, only for the `BearerTokenConfig` in here?

Reply:
As we have used the type from client.config, we can only override them using a subset of original union type

Reply:
https://github.com/ballerina-platform/module-ballerinax-client.config/blob/4bcdc36f4ebfcbbc2d8628c0a161b2ffa6ed82b6/config_types.bal#L55